src/compas_fab/robots/constraints.py

```diff
@@ -329,3 +332,116 @@ def __repr__(self):
+
+    NOTE: getMarkers() could be used to visualize feedback?
+    http://docs.ros.org/kinetic/api/moveit_core/html/classkinematic__constraints_1_1VisibilityConstraint.html#a28a578edc7a97627aaa78ef061871072
+
+    Attributes
+    ----------
+    target_frame: :class:`compas.geometry.Frame`
+        The frame of the disc; as the robot moves, the frame of the disc may change as well
+        this can be in the frame of a particular robot link, for example
+    target_radius: float
+        The radius of the disc that should be maintained visible
+    sensor_frame: :class:`compas.geometry.Frame`
+        The frame in which visibility is to be maintained.
+        Frame id defaults to robot base link.
+        It is assumed the sensor can look directly at the target, in any direction.
+        This assumption is usually not true, but additional PositionConstraints
+        can resolve this issue.
+    target_frame_reference_link: string, optional
+        The name of the link that the target_frame is associated with.
+        Defaults to base link, thus world coordinates
+    sensor_frame_reference_link: string, optional
+        The name of the link that the sensor_frame is associated with.
+        Defaults to base link, thus world coordinates
+    cone_sides: int, optional
+        From the sensor origin towards the target, the disc forms a visibility cone.
+        This cone is approximated using many sides. For example, when using 4 sides,
+        that in fact makes the visibility region be a pyramid.
+        This value should always be 3 or more.
+        Defaults to 3
+    max_view_angle: float, optional
+        Even though the disc is maintained visible, the visibility cone can be very small
+        because of the orientation of the disc with respect to the sensor. It is possible
+        for example to view the disk almost from a side, in which case the visibility cone
+        can end up having close to 0 volume. The view angle is defined to be the angle between
+        the normal to the visibility disc and the direction vector from the sensor origin.
+        The value below represents the minimum desired view angle. For a perfect view,
+        this value will be 0 (the two vectors are exact opposites). For a completely obstructed view
+        this value will be Pi/2 (the vectors are perpendicular). This value defined below
+        is the maximum view angle to be maintained. This should be a value in the open interval
+        (0, Pi/2). If 0 is set, the view angle is NOT enforced.
+        Defaults to 0.0
+    max_range_angle: float, optional
+        This angle is used similarly to max_view_angle but limits the maximum angle
+        between the sensor origin direction vector and the axis that connects the
+        sensor origin to the target frame origin. The value is again in the range (0, Pi/2)
+        and is NOT enforced if set to 0.
+        Defaults to 0.0
+    sensor_view_direction: int, optional
```

Comment:
probably needs to be defined and checked.
see possible_types of `BoundingVolume` check